webui/react/src/hooks/useModal/Checkpoint/useModalCheckpointRegister.tsx

```diff
@@ -78,40 +79,68 @@ const useModalCheckpointRegister = ({ onClose }: Props = {}): ModalHooks => {
   const registerModelVersion = useCallback(async (state: ModalState) => {
     const {
       selectedModelName, versionDescription, tags,
-      metadata, versionName, checkpointUuid,
+      metadata, versionName, checkpoints,
     } = state;
-    if (!selectedModelName || !checkpointUuid) return;
+    if (!selectedModelName || !checkpoints) return;
     try {
-      const response = await postModelVersion({
-        body: {
-          checkpointUuid,
-          comment: versionDescription,
-          labels: tags,
-          metadata,
+      if (checkpoints.length === 1) {
+        const response = await postModelVersion({
+          body: {
+            checkpointUuid: checkpoints[0],
+            comment: versionDescription,
+            labels: tags,
+            metadata,
+            modelName: selectedModelName,
+            name: versionName,
+          },
           modelName: selectedModelName,
-          name: versionName,
-        },
-        modelName: selectedModelName,
-      });
-
-      if (!response) return;
-
-      modalClose(ModalCloseReason.Ok);
-
-      notification.open({
-        btn: null,
-        description: (
-          <div className={css.toast}>
-            <p>{`"${versionName || `Version ${selectedModelNumVersions + 1}`}"`} registered</p>
-            <Link path={paths.modelVersionDetails(selectedModelName, response.id)}>
-              View Model Version
-            </Link>
-          </div>),
-        message: '',
-      });
+        });
+
+        if (!response) return;
+
+        modalClose(ModalCloseReason.Ok);
+
+        notification.open({
+          btn: null,
+          description: (
+            <div className={css.toast}>
+              <p>{`"${versionName || `Version ${selectedModelNumVersions + 1}`}"`} registered</p>
+              <Link path={paths.modelVersionDetails(selectedModelName, response.id)}>
+                View Model Version
+              </Link>
+            </div>),
+          message: '',
+        });
+      } else {
+        for (const checkpointUuid of checkpoints) {
+          await postModelVersion({
+            body: {
+              checkpointUuid,
+              comment: versionDescription,
+              labels: tags,
+              metadata,
+              modelName: selectedModelName,
+            },
+            modelName: selectedModelName,
+          });
+        }
+        modalClose(ModalCloseReason.Ok);
+
+        notification.open({
+          btn: null,
+          description: (
+            <div className={css.toast}>
+              <p>{checkpoints.length} versions registered</p>
+              <Link path={paths.modelDetails(selectedModelName)}>
+                View Model
+              </Link>
+            </div>),
+          message: '',
+        });
+      }
     } catch (e) {
       handleError(e, {
-        publicSubject: 'Unable to register checkpoint.',
+        publicSubject: `Unable to register ${pluralizer(checkpoints.length, 'checkpoint')}.`,
```

Comment:
Minor nit: I am not seeing this getting hit for errors (which is another ticket), but I think this would be misleading in case of partial failures